src/Dropdown.elm

```diff
@@ -126,23 +127,75 @@ init id =
     Dropdown.basic DropdownMsg OptionPicked Element.text Element.text
 
 -}
-basic : (Msg item -> msg) -> (Maybe item -> msg) -> (item -> Element msg) -> (Bool -> Bool -> item -> Element msg) -> Config item msg
+basic :
+    (Msg item -> msg)
+    -> (Maybe item -> msg)
+    -> (item -> Element msg)
+    -> (Bool -> Bool -> item -> Element msg)
+    -> Config item msg
 basic dropdownMsg onSelectMsg itemToPrompt itemToElement =
     Config
-        { dropdownType = Basic
-        , promptElement = el [ width fill ] (text "-- Select --")
-        , filterPlaceholder = Nothing
-        , dropdownMsg = dropdownMsg
-        , onSelectMsg = onSelectMsg
+        { closeButton = text "▲"
         , containerAttributes = []
-        , selectAttributes = []
+        , dropdownMsg = dropdownMsg
+        , dropdownType = Basic
+        , filterPlaceholder = Nothing
+        , itemToElement = itemToElement
+        , itemToPrompt = itemToPrompt
+        , itemToText = \_ -> ""
         , listAttributes = []
+        , onSelectMsg = onSelectMsg
+        , openButton = text "▼"
+        , promptElement = el [ width fill ] (text "-- Select --")
         , searchAttributes = []
-        , itemToPrompt = itemToPrompt
+        , selectAttributes = []
+        }
+
+
+{-| Create a basic dropdown with custom configuration. This takes:
+
+    - The message to wrap all the internal messages of the dropdown
+    - A message to trigger when an item is selected
+    - A function to get the Element to display from an item, to be used in the select part of the dropdown
+    - A function to get the Element to display from an item, to be used in the item list of the dropdown
+    - An element to display as close button
+    - An element to display as open button
+    - An element to be used in the prompt
+    - A list of attributes to be passed to the container element.
+    - A list of attributes to be passed to the list element.
+    - A list of attributes to be passed to the search element.
+    - A list of attributes to be passed to the select element.
+
+-}
+custom :
+    (Msg item -> msg)
+    -> (Maybe item -> msg)
+    -> (item -> Element msg)
+    -> (Bool -> Bool -> item -> Element msg)
+    -> Element msg
+    -> Element msg
+    -> Element msg
+    -> List (Attribute msg)
+    -> List (Attribute msg)
+    -> List (Attribute msg)
+    -> List (Attribute msg)
+    -> Config item msg
+custom dropdownMsg onSelectMsg itemToPrompt itemToElement closeBtn openBtn promptElement containerAttributes listAttributes searchAttributes selectAttributes =
+    Config
+        { closeButton = closeBtn
```

Comment:
Hey there, thanks for your work! Considering the number of arguments here and that many of them have the same type I'd change it to single record

Reply:
This makes total sense, thanks for the suggestion 😉